### PR TITLE
[TACHYON-1428] Added checkstyle rule for documentation of public methods and classes

### PR DESCRIPTION
**Pull request by HorizonNet**
https://tachyon.atlassian.net/browse/TACHYON-1428

### CODE REVIEW

File: build/checkstyle/suppressions.xml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+<!DOCTYPE suppressions PUBLIC "-//Puppy Crawl//DTD Suppressions 1.1//EN"
+        "http://www.puppycrawl.com/dtds/suppressions_1_1.dtd">
+<suppressions>
+    <suppress files="[\\/]src[\\/]test[\\/].*" id="MethodNameRegular" />
+    <suppress files="[\\/]src[\\/]main[\\/].*" id="MethodNameTest" />
+</suppressions>
```
Comment (jsimsa) [inline build/checkstyle/suppressions.xml]: add a missing new line

File: build/checkstyle/tachyon_checks.xml [hunk 2]
```diff
@@ -223,6 +223,14 @@
     </module>
     <module name="MissingDeprecated"/>
     <module name="EqualsHashCode"/>
+    <module name="JavadocMethod">
```
Comment (apc999) [inline build/checkstyle/tachyon_checks.xml]: @HorizonNet , could you add some comment here to explain this rule and the rule below ?
We should keep the checks config file well documented, so we will try to comment other modules too.

File: build/checkstyle/suppressions.xml (hunk 1 above)
Comment (apc999) [inline build/checkstyle/suppressions.xml]: also, could you explain each suppress rule in comment, so we know what is suppressed for what?

Thanks

**Comment by apc999** [comment]
@HorizonNet thanks for the awesome work and actually all the effort before this PR to fix/add javadoc in this project in order to enable this stylecheck. It turns out really exciting for me. Thanks so much!

I and @jsimsa left a few comments, mostly asking for some comments in the config. After that, we will merge the PR. Thanks!

**Comment by HorizonNet** [comment]
@apc999 Thanks for the comments. I updated the PR with some comments about the suppressions and the rules. I also changed the name of the suppressions to be more meaningful.

**Comment by apc999** [comment]
@HorizonNet thanks! This is awesome!
+1 LGTM

**Comment by jsimsa** [comment]
LGTM

**Pull request merged**